// ngx-echarts/esm5/core/lib/ngx-echarts-core.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { NgxEchartsCoreDirective } from './ngx-echarts-core.directive';
import * as ɵngcc0 from '@angular/core';
var NgxEchartsCoreModule = /** @class */ (function () {
    function NgxEchartsCoreModule() {
    }
NgxEchartsCoreModule.ɵfac = function NgxEchartsCoreModule_Factory(t) { return new (t || NgxEchartsCoreModule)(); };
NgxEchartsCoreModule.ɵmod = /*@__PURE__*/ ɵngcc0.ɵɵdefineNgModule({ type: NgxEchartsCoreModule });
NgxEchartsCoreModule.ɵinj = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjector({ imports: [[]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NgxEchartsCoreModule, [{
        type: NgModule,
        args: [{
                imports: [],
                declarations: [
                    NgxEchartsCoreDirective
                ],
                exports: [
                    NgxEchartsCoreDirective
                ]
            }]
    }], function () { return []; }, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgxEchartsCoreModule, { declarations: function () { return [NgxEchartsCoreDirective]; }, exports: function () { return [NgxEchartsCoreDirective]; } }); })();
    return NgxEchartsCoreModule;
}());
export { NgxEchartsCoreModule };
