// ngx-echarts/esm5/lib/ngx-echarts.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { NgxEchartsDirective } from './ngx-echarts.directive';
import * as ɵngcc0 from '@angular/core';
var NgxEchartsModule = /** @class */ (function () {
    function NgxEchartsModule() {
    }
NgxEchartsModule.ɵfac = function NgxEchartsModule_Factory(t) { return new (t || NgxEchartsModule)(); };
NgxEchartsModule.ɵmod = /*@__PURE__*/ ɵngcc0.ɵɵdefineNgModule({ type: NgxEchartsModule });
NgxEchartsModule.ɵinj = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjector({ imports: [[]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NgxEchartsModule, [{
        type: NgModule,
        args: [{
                imports: [],
                declarations: [
                    NgxEchartsDirective
                ],
                exports: [
                    NgxEchartsDirective
                ]
            }]
    }], function () { return []; }, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgxEchartsModule, { declarations: function () { return [NgxEchartsDirective]; }, exports: function () { return [NgxEchartsDirective]; } }); })();
    return NgxEchartsModule;
}());
export { NgxEchartsModule };
