// angular2-ladda/esm5/lib/ladda.module.js
import { __decorate } from "tslib";
import { NgModule } from '@angular/core';
import { LaddaDirective } from './ladda.directive';
import { LaddaConfig } from './ladda-config';
import * as ɵngcc0 from '@angular/core';
var LaddaModule = /** @class */ (function () {
    function LaddaModule() {
    }
    LaddaModule_1 = LaddaModule;
    LaddaModule.forRoot = function (config) {
        return {
            ngModule: LaddaModule_1,
            providers: [
                { provide: LaddaConfig, useValue: config }
            ]
        };
    };
    var LaddaModule_1;
LaddaModule.ɵfac = function LaddaModule_Factory(t) { return new (t || LaddaModule)(); };
LaddaModule.ɵmod = /*@__PURE__*/ ɵngcc0.ɵɵdefineNgModule({ type: LaddaModule });
LaddaModule.ɵinj = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjector({});
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(LaddaModule, [{
        type: NgModule,
        args: [{
                declarations: [LaddaDirective],
                exports: [LaddaDirective]
            }]
    }], function () { return []; }, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(LaddaModule, { declarations: function () { return [LaddaDirective]; }, exports: function () { return [LaddaDirective]; } }); })();
    return LaddaModule;
}());
export { LaddaModule };
